File: src/main/kotlin/com/chopeks/parser/SquirrelParserUtil.java
package com.chopeks.parser;

import com.intellij.lang.PsiBuilder;
import com.intellij.lang.parser.GeneratedParserUtilBase;
import com.intellij.psi.tree.IElementType;
import org.jetbrains.annotations.NotNull;

import static com.chopeks.SquirrelTokenTypes.*;
import static com.chopeks.SquirrelTokenTypesSets.COMMENTS;
import static com.chopeks.SquirrelTokenTypesSets.WHITE_SPACES;

public class SquirrelParserUtil extends GeneratedParserUtilBase {

    public static boolean prevIsBrace(@NotNull PsiBuilder builder_, int level) {
        IElementType type = null;
        for (int i = 1; i < builder_.getCurrentOffset(); i++) {
            type = builder_.rawLookup(-i);
            if (!COMMENTS.contains(type) && !WHITE_SPACES.contains(type)) {
                break;
            }
        }

        return (type == RBRACE || type == SEMICOLON || type == SEMICOLON_SYNTHETIC);
    }
}

File: src/main/kotlin/com/chopeks/parser/SquirrelParserUtil.kt
package com.chopeks.parser

import com.chopeks.SquirrelTokenTypes
import com.chopeks.SquirrelTokenTypesSets
import com.intellij.lang.PsiBuilder
import com.intellij.lang.parser.GeneratedParserUtilBase
import com.intellij.psi.tree.IElementType

object SquirrelParserUtil : GeneratedParserUtilBase() {
	@JvmStatic
	fun prevIsBrace(builder: PsiBuilder, level: Int): Boolean {
		var type: IElementType? = null
		for (i in 1..<builder.currentOffset) {
			type = builder.rawLookup(-i)
			if (!SquirrelTokenTypesSets.COMMENTS.contains(type) && !SquirrelTokenTypesSets.WHITE_SPACES.contains(type)) {
				break
			}
		}

		return (type === SquirrelTokenTypes.RBRACE || type === SquirrelTokenTypes.SEMICOLON || type === SquirrelTokenTypes.SEMICOLON_SYNTHETIC)
	}
}